jlm/llvm/backend/rvsdg2jlm/rvsdg2jlm.cpp

```diff
@@ -410,12 +411,12 @@ convert_theta_node(const rvsdg::Node & node, context & ctx)
 static inline void
 convert_lambda_node(const rvsdg::Node & node, context & ctx)
 {
-  JLM_ASSERT(is<lambda::operation>(&node));
-  auto lambda = static_cast<const lambda::node *>(&node);
+  JLM_ASSERT(is<llvm::LlvmLambdaOperation>(&node));
+  auto lambda = static_cast<const rvsdg::LambdaNode *>(&node);
   auto & module = ctx.module();
   auto & clg = module.ipgraph();
 
-  const auto & op = lambda->GetOperation();
+  const auto & op = dynamic_cast<llvm::LlvmLambdaOperation &>(lambda->GetOperation());
```

Comment:
This is now what I do not like. We get these casts everywhere where we need the more specialized form. Any change we could introduce a templated `GetOperation<>()` method that tries to cast it down to the concrete LambdaOperation type?